team-validator.js

```diff
@@ -105,7 +105,13 @@ class Validator {
 		if (!template.exists) {
 			return ["The Pokemon '" + set.species + "' does not exist."];
 		}
-		set.species = template.species;
```

Comment:
Apparently I didn't research enough, but removing this line is enough without all the other stuff...